src/ase/cw/view/ServerFrame.java

```diff
@@ -72,47 +81,63 @@ public int getServerId() {
     public void actionPerformed(ActionEvent e) {
 
         if (e.getSource() == breakButton) {
-            logButtonPress(breakButton);
-            breakButton.setEnabled(false);
-            restartButton.setEnabled(true);
-            oc.pauseOrderProcess(this.serverId);
+            if(pausable!=null) {
+                logButtonPress(breakButton);
+                breakButton.setEnabled(false);
+                restartButton.setEnabled(true);
+                pausable.pause();
+            }
+
         }
 
         if (e.getSource() == restartButton) {
-            logButtonPress(restartButton);
-            restartButton.setEnabled(false);
-            breakButton.setEnabled(true);
-            oc.restartOrderProcess(this.serverId);
+            if(pausable!=null){
+                logButtonPress(restartButton);
+                restartButton.setEnabled(false);
+                breakButton.setEnabled(true);
+                pausable.unPause();
+            }
         }
     }
 
     @Override
     public void updateView(OrderConsumer server, Order order) {
-        this.textArea.setText(
-                "Status: "+(server.getStatus()
-                        +"\nOrder: "+order.getCustomerId()+ (order.isPriorityOrder() ? " (priority)" : "")
-                        +"\nItems: "+order.getOrderItems().size()
-                        +"\nSubtotal: £"+order.getBill().getSubtotal()
-                        +"\nTotal: £"+order.getBill().getTotal()
-                ));
+
+        this.statusArea.setText("Status: " + (server.getStatus()));
+        if(server.getStatus()== ServerStatusEnum.ServerStatus.STOPPED){
+            this.dispose();
+        } else if (server.getStatus() == ServerStatusEnum.ServerStatus.BUSY && order != null) {
+            this.textArea.setText(
+                             "Order: " + order.getCustomerId()
+                            + "\nItems: " + order.getOrderItems().size()
+                            + "\nSubtotal: £" + order.getBill().getSubtotal()
+                            + "\nTotal: £" + order.getBill().getTotal()
+                    );
+        } else if(server.getStatus()== ServerStatusEnum.ServerStatus.FREE || server.getStatus()== ServerStatusEnum.ServerStatus.PAUSED){
+            this.textArea.setText("");
+        } else {
+
```

Comment:
I am not fan of the ServerFrame deciding what to display to the user based on status

Reply:
Why?
You did exactly the same? You only did it in two different methods (One with a order parameter one without).
There is nothing wrong with this solution. Otherwise please explain why this is bad.

Reply:
the reason I had two methods is because when the Server is free or paused there should be no Order passed as a parameter as there should be no Order associated with the Server - the current order processing design does not account for this correctly. this is a workaround that should not be necessary